Sources/swift-format/Frontend/Frontend.swift

```diff
@@ -241,6 +243,61 @@ class Frontend {
       }
     }
 
+    // Load global configuration file
+    // First URLs are created, then they are queried. First match is loaded
+    var configLocations: [URL?] = []
+
+    #if os(Windows)
+      if let localAppData = ProcessInfo.processInfo.environment["LOCALAPPDATA"] {
+        configLocations.append(URL(fileURLWithPath: localAppData))
+      }
+      if let programData = ProcessInfo.processInfo.environment["PROGRAMDATA"] {
+        configLocations.append(URL(fileURLWithPath: programData))
+      }
+    #else
+      if let xdgConfigHome = ProcessInfo.processInfo.environment["XDG_CONFIG_HOME"] {
+        configLocations.append(URL(fileURLWithPath: xdgConfigHome))
+      }
+
+      if let libraryUrl = FileManager.default.urls(
+        for: .applicationSupportDirectory, in: .userDomainMask
+      ).first {
+        configLocations.append(libraryUrl)
+      }
+
+      if let xdgConfigDirs = ProcessInfo.processInfo.environment["XDG_CONFIG_DIRS"] {
+        configLocations += xdgConfigDirs.split(separator: ":").map { xdgConfigDir in
+          URL(fileURLWithPath: String(xdgConfigDir))
+        }
+      }
+
+      if let libraryUrl = FileManager.default.urls(
+        for: .applicationSupportDirectory, in: .systemDomainMask
+      ).first {
+        configLocations.append(libraryUrl)
+      }
+    #endif
+
+    for case var location? in configLocations {
+      if #available(macOS 13.0, iOS 16.0, *) {
+        location.append(components: "swift-format", "config.json")
```

Comment:
This will never be true on non-Apple platforms. I would prefer to just have one branch here that uses `appendPathComponent`. We do that in many other places as well. Just has lower risk of platforms diverging.

Reply:
I was a confused about how language features availability can be checked in swift. In the documentation they are listed as available/deprecated by OS and Xcode version but no mention of a swift version which would be arguably better for usage in non-apple environment.
I'll change it back to `appendPathComponent`. Was trying to avoid it if possible because of the note in the [documentation](https://developer.apple.com/documentation/foundation/url/1780239-appendingpathcomponent).

Reply:
I think we might be able to switch to `append(components:)` if we consistently use [swift-foundation](https://github.com/swiftlang/swift-foundation) instead of [swift-corelibs-foundation](https://github.com/swiftlang/swift-corelibs-foundation) on non-Apple platforms but we don’t right now because we need to support building with Swift 5.10, which doesn’t have the new swift-foundation yet.